lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -212,7 +212,7 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
                              scf::YieldOp, scf::WhileOp, scf::ForOp,
                              /// memref
                              memref::AllocOp, memref::AllocaOp, memref::LoadOp,
-                             memref::StoreOp,
+                             memref::StoreOp, memref::CopyOp,
```

Comment:
If you don't make this change, does the pass fail with the below error of `"Unhandled operation during BuildOpGroups()"`? I think that is reasonable behavior--the pass doesn't handle this op, and expects an earlier pass in the compiler pipeline to handle it.

Reply:
_Maybe_ in the `Default` case, you could add some conditional logic like

```
if (isa<memref::CopyOp>(op))
  error.addNote("Please run --handshake-legalize-memrefs")
```

Reply:
But even here, I don't like the idea of naming a specific pass--that will inevitably get out of date. And, we should be providing a correct pass pipeline in a tool like `hlstool` regardless.

Reply:
> If you don't make this change, does the pass fail with the below error of "Unhandled operation during BuildOpGroups()"?

Yes, that what will happen.

> But even here, I don't like the idea of naming a specific pass--that will inevitably get out of date. And, we should be providing a correct pass pipeline in a tool like hlstool regardless.

I agree. My thought was just: since we want to lower from Calyx to dialects like `hw`, `seq`,`comb`, and eventually `sv` to have System Verilog, it's unnatural to think of a pass provided by `handshake`. So I wanted to remind the future developers that there's actually an available conversion by emitting this error message.

Reply:
I think the current error message is good enough for future compiler developers. In general, error messages are to help end users diagnose problems in their input: https://llvm.org/docs/CodingStandards.html#error-and-warning-messages. I don't think we need a custom error message for the situation where a compiler developer constructed the wrong compiler pipeline.